source/ftrack_connect_pipeline_qt/ui/asset_manager/asset_manager.py

```diff
@@ -768,8 +820,21 @@ def on_collapse(self, collapsed):
                 self._component_and_version_widget.set_component_filename(
                     self._component_path
                 )
+
+                # Find other compatible versions
+                query = (
+                    'select is_latest_version, id, asset, components, components.name, '
+                    'components.id, version, asset , asset.name, asset.type.name from '
+                    'AssetVersion where asset.id is "{}" and components.name is "{}"'
+                    'order by version ascending'
+                ).format(
+                    self._asset_id,
+                    self._component_name,
+                )
+                versions = self.session.query(query).all()
+
                 self._component_and_version_widget.set_version(
-                    self._version_nr, versions=self._versions_collection
+                    self._version_nr, versions=versions
```

Comment:
The main idea is to not have core code in qt. 
So maybe I'm wrong, but is it possible that we were using self._versions_collection because we were passing the versions from the core repository and not querying them in the qt repo? If so I highly recommend to change it back so we keep all the logic in core and not in qt

Reply:
Not sure I follow, is this related to the same issue as with opener - we need to move logic out of QT to core, e.g. put the versions query in a plugin and have it called by QT? Or just bringing back variable self._versions_collection? 

When I look at the original code, self._versions_collection does not exists in the core and there is no function to populate it?